File: C,C++/p11332.cpp
#include <iostream>

using namespace std;

int sumDigit(int);

int main() {
    int input;
    while (true) {
        cin >> input;
        if (input == 0) break;
        else {
            int sum = input;
            while (true) {
                sum = sumDigit(sum);
                if (sum < 10) break;
            }
            cout << sum << endl;
        }
    }
    
    return 0;
}

int sumDigit(int num) {
    int res = 0;
    
    while (num > 9) {
        res += num % 10;
        num /= 10;
    }
    
    res += num;
    
    return res;
}

File: Java/p11332.java

import java.util.Scanner;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author DarKnight98
 */
public class p11332 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        while (true) {
            int num = input.nextInt();
            if (num == 0) break;
            int total = Integer.MAX_VALUE;
            while (total > 9) {
                num = sum(num);
                total = num;
            }
            System.out.println(total);
        }
    }
    
    static int sum (int num) {
        int temp = num;
        int total = 0;
        while (temp > 9) {
            total += temp%10;
            temp /= 10;
        }
        total += temp;
        return total;
    }
}
